### DISCUSSION TITLE
Can't get my custom 500 page to work

### DISCUSSION THREAD

**Discussion by pgbradbury** [vercel/next.js#47501] [Help]
### Summary

I have built a custom 500 page and named it 500.jsx (and also tried 500.js) and placed it in the pages directory.

I have an _error.jsx page in the pages directory as well.

When I trigger an error, the page that is rendered is the default next error page for a 500 error.

I am running next 13.2.4 and react 18.2.0

I can see the error page render correctly if I go directly to it at /500

Any help greatly appreciated as I am at a loss.

### Additional information

```js
//_error.jsx

import * as Sentry from "@sentry/nextjs"
import NextErrorComponent from "next/error"

const CustomErrorComponent = (props) => {
    // If you're using a Nextjs version prior to 12.2.1, uncomment this to
    // compensate for https://github.com/vercel/next.js/issues/8592
    Sentry.captureUnderscoreErrorException(props)

    return <NextErrorComponent statusCode={props.statusCode} />
}

CustomErrorComponent.getInitialProps = async (contextData) => {
    // In case this is running in a serverless function, await this in order to give Sentry
    // time to send the error before the lambda exits
    await Sentry.captureUnderscoreErrorException(contextData)

    // This will contain the status code of the response
    return NextErrorComponent.getInitialProps(contextData)
}

export default CustomErrorComponent
```

**Comment by icyJoseph** [comment]
> I can see the error page render correctly if I go directly to it at /500

That's because you don't have a 404 page, so it uses the _error.

> 500 errors are handled both client-side and server-side by the Error component. If you wish to override it, define the file pages/_error.js and add the following code:

Do you have a getInitialProps handler in `_app`?

I think without `_app` getInitialProps, a 500 pages shows over _error, but if there's `_app` getInitialProps, then you'd see the 500 for a frame and then the `_error` page. As far as I remember this applies for errors during rendering and during something like GetServerSideProps.

The rules of what goes where here are a bit difficult, I think anyway. If you could share some more code to reproduce this so that we can understand what's going on, that'd be awesome.

**Reply by pgbradbury** [reply]
Thanks for taking a look. I don't have a handler in my _app.jsx (code below). I guess I am confused, as my understanding was if you had a 500.js file in pages it would use that instead of rendering the default. All the _error.js does is log the error with Sentry and then use the Next Error Component to deal with rendering the error. I was referring to seeing the 500 page OK if I went directly to it, to rule out that that page was erroring itself and causing the issue.

So this is the _app.jsx

```javascript
//_app.jsx

import { useEffect } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import PropTypes from "prop-types"
import CssBaseline from "@mui/material/CssBaseline"
import AdapterDateFns from "@mui/lab/AdapterDateFns"
import LocalizationProvider from "@mui/lab/LocalizationProvider"
import { ThemeProvider } from "@mui/material/styles"
import { CacheProvider } from "@emotion/react"
import { SessionProvider, useSession, signIn } from "next-auth/react"
import { RecoilRoot, useRecoilState } from "recoil"
import { setUser } from "@sentry/nextjs"
import { ReactQueryDevtools } from "react-query/devtools"

import radiateTheme from "../app/components/radiateTheme"
import createEmotionCache from "../app/libraries/createEmotionCache"
import Loading from "../app/components/elements/Loading"
import { organisationIdAtom } from "../app/state/atoms"

// Client-side cache, shared for the whole session of the user in the browser.
const clientSideEmotionCache = createEmotionCache()
const queryClient = new QueryClient()

export default function MyApp(props) {
    const {
        Component,
        emotionCache = clientSideEmotionCache,
        pageProps: { session, ...pageProps },
    } = props

    return (
        <SessionProvider session={session}>
            <CacheProvider value={emotionCache}>
                <ThemeProvider theme={radiateTheme}>
                    <LocalizationProvider dateAdapter={AdapterDateFns}>
                        <RecoilRoot>
                            <QueryClientProvider client={queryClient}>
                                <CssBaseline />
                                <ReactQueryDevtools initialIsOpen={true} />
                                {Component.secure ? (
                                    <Auth>
                                        <Component {...pageProps} />
                                    </Auth>
                                ) : (
                                    <Component {...pageProps} />
                                )}
                            </QueryClientProvider>
                        </RecoilRoot>
                    </LocalizationProvider>
                </ThemeProvider>
            </CacheProvider>
        </SessionProvider>
    )
}

function Auth({ children }) {
    const { data: session, status } = useSession()
    const isUser = !!session?.user
    const [organisationId, setOrganisationId] = useRecoilState(organisationIdAtom)

    useEffect(() => {
        if (status === "loading") return // Do nothing while loading
        if (!isUser) signIn() // If not authenticated, force log in
    }, [isUser, status])

    if (isUser) {
        setUser({ email: session.user.email }) // Attach user email to all Sentry reports
        if (organisationId == 0) {
            setOrganisationId(session.user.organisation_id)
        }
        return children
    }

    return <Loading />
}

MyApp.propTypes = {
    Component: PropTypes.elementType.isRequired,
    emotionCache: PropTypes.object,
    pageProps: PropTypes.object.isRequired,
}
```

**Reply by pgbradbury** [reply]
Is it that the Next Error Component doesn't look for the 500 error page if called from the _error.jsx page? So I need to check the status code and call my custom page directly inside the _error.jsx code instead of the Next Error Component?

**Reply by icyJoseph** [reply]
I'm out of office now but I'll take a look when I get some time. In general it's best to test these in isolation. Like, make a clean project and try out different configurations.

**Comment by pgbradbury** [comment]
Has anyone any further thoughts on this, still no further with it?

**Reply by alwaysmonodeep** [reply]
i have same issues. any solution?

**Comment by DonikaV** [comment]
I have question, i return from Server Api route status 500, and in my component i am throwing error, but it doesnt redirect to 500 page. How i can redirect to error.js page if i have status 500 on my request?

**Comment by droidada** [comment]
I am having this same issue. Who's found a way to resolve this please?

**Reply by alwaysmonodeep** [reply]
i am with same issue. solution found?

**Comment by ValentinH** [comment]
3 years later, I still can't reliably show a clean error page in production (Vercel / Pages router). I feel like I need a PHD to be able to just render a clean branded error page (Bonus point for having it translated) 😅

**Reply by icyJoseph** [reply]
Have you tried recently with the global-error page? https://nextjs.org/docs/app/getting-started/error-handling#global-errors assuming App Router of course

**Reply by ValentinH** [reply]
I'm on the Pages Router 🥲
